File: 230-kth-smallest-element-in-a-bst/230-kth-smallest-element-in-a-bst.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int KthSmallest(TreeNode root, int k) {
         List<int> nums = inorder(root, new List<int>());
            return nums.ToArray()[k-1];
    }
    
    
    public List<int> inorder(TreeNode root, List<int> arr) {
            if (root == null) 
                return arr;
            
            inorder(root.left, arr);
            arr.Add(root.val);
            inorder(root.right, arr);
        
            return arr;
       }
}

File: 230.kth-smallest-element-in-a-bst.java
/*
 * @lc app=leetcode id=230 lang=java
 *
 * [230] Kth Smallest Element in a BST
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int kthSmallest(TreeNode root, int k) {
        
        ArrayList<Integer> myList = inOrder(root, new ArrayList<Integer>());

        return myList.get(k-1);

    }

    private ArrayList<Integer> inOrder(TreeNode root, ArrayList<Integer> result)
    {
        if(root == null)
        return result;

        inOrder(root.left,result);
        result.add(root.val);
        inOrder(root.right,result);

        return result;
    }
}
// @lc code=end

